keras/layers/core.py

```diff
@@ -372,24 +372,14 @@ def _fix_unknown_dimension(self, input_shape, output_shape):
         return tuple(output_shape)
 
     def compute_output_shape(self, input_shape):
-        return (input_shape[0],) + self._fix_unknown_dimension(
-            input_shape[1:], self.target_shape)
+        if None in input_shape[1:]:
+            return (input_shape[0],) + tuple(s if s != -1 else None for s in self.target_shape)
+        else:
+            return (input_shape[0],) + self._fix_unknown_dimension(
```

Comment:
Since this function is only called once, consider in-lining it (not saying you *should* do it, but consider whether it would improve the code if you did it -- maybe it would).

Reply:
I had considered that, but I figured keeping it separate would be better. That way it can potentially be used by layers that subclass `Reshape` (or other code in general). In addition it is more in line with the "separation of concerns" philosophy.